lib/at_monitor/at_monitor.c

```diff
@@ -26,11 +26,14 @@ static K_FIFO_DEFINE(at_monitor_fifo);
 static K_HEAP_DEFINE(at_monitor_heap, CONFIG_AT_MONITOR_HEAP_SIZE);
 static K_WORK_DEFINE(at_monitor_work, at_monitor_task);
 
```

Comment:
Not related to this line but I'll add this comment somewhere to get a discussion thread for it. This line in `CMakeLists.txt` of at_monitor doesn't work for `native_posix` due to a linker bug:
`zephyr_linker_sources(RWDATA at_monitor.ld)`

And due to this, tests won't work in this PR branch. I've done few hacks into at_monitor, sms library and tests to get around this to be able to develop the tests for nrf_modem_at.h.

The bug should be handled later in Zephyr side. This is what @tejlmand commented about it:
https://github.com/zephyrproject-rtos/zephyr/blob/e4234aeb89c26047256e78707cf8084d8c571132/cmake/extensions.cmake#L1161-L1177
seems it is intended that NOINIT, RWDATA, RODATA, ROM_START should be placed in predefined section, which means the native_posix ld template is wrong on this.

Reply:
Fixed here: https://github.com/zephyrproject-rtos/zephyr/pull/41644

Reply:
and `[nrf fromlist]` here: https://github.com/nrfconnect/sdk-zephyr/pull/685

Please make a manifest update in this PR with `sdk-zephyr` using revision `pull/685/head`.

Reply:
Manifest update made and everything works fine so this is fixed now.